.github/workflows/ci.yml

```
@@ -292,6 +292,103 @@ jobs:
             dist-newstyle
           key: ${{ env.CABAL_CACHE_VERSION }}-cabal-${{ matrix.os }}-${{ matrix.ghc }}-${{ hashFiles(format('cabal.GHC-{0}.config', matrix.ghc)) }}-${{ github.sha }}
 
+  build-docs:
+    name: "Build and deploy documentation to GitHub pages"
+    needs: [coverage]
+    runs-on: ${{ matrix.os }}
```

This action's going to need the checks from line 548, isn't it?

---

although it would be nice if the doc _build_ still ran...

---

I don't believe it will, with the alternative action we are using to manage the deployment / how it authenticates.

---

What do you expect it to do? ISTM that from a random user's fork the choices are "don't run", "fail at the push step", and "silently do nothing at the push step if the keys don't work" and I doubt the last is what we want even if it's an available choice. Maybe it's already got equivalent check logic in it though?

---

After reading more about [the authentication method](https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#about-the-github_token-secret), I think you're right that we still need the check, since by default this token only has read permission when it is generated in the context of a pull request from a public fork (see [here](https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#permissions-for-the-github_token)).

I wrongly assumed that the token had the same permissions regardless of PR source; I'll add the appropriate checks back (and, to make sure the doc build still runs, I can split the process into separate jobs so only the deployment part is gated).